### DISCUSSION TITLE
make Combobox `optionTemplate` prop generic

### DISCUSSION THREAD

**Discussion by vankhoawin** [twilio-labs/paste#1012] [Ideas] [answered]
https://github.com/twilio-labs/paste/blob/91903a6b7fe2682b6c132722a6e82aa141ec497f/packages/paste-core/components/combobox/src/types.ts#L22

Currently, `optionTemplate` is a function with a `string | {}` input. We can provide stronger and more accurate typing by making ComboboxProps and furthermore [Combobox](https://github.com/twilio-labs/paste/blob/84dd1a4aea0410f4e2de844538f60add6a3931b1/packages/paste-core/components/combobox/src/Combobox.tsx#L143) generic.

some examples how to do so in StackOverflow: https://stackoverflow.com/a/58473012

**Comment by TheSisb** [comment, marked as answer]
Hi @vnguyen94, good suggestion! Did you encounter a specific issue with the typing or is this more of an observation where we can make an enhancement?

Always open to PRs too <3

**Reply by vankhoawin** [reply]
I guess, both? 😄 we ran into it when trying to type `item` as `SomeInterface/Type`, and it wouldn't let us without using `@ts-ignore` to bypass that.

It's tweek so just wanted to leave a placeholder before I forgot, maybe I can circle back on a PR afterwards

**Reply by corinne-nagel** [reply]
Hey, @vnguyen94 I'm closing out discussions that have been finished/answered. Did you get a chance to PR this? 🙂

**Reply by vankhoawin** [reply]
i did not 😓  kinda getting busy lately, but the issue still exists. a look at antd has a workaround for typing `React.forwardRef` components: https://github.com/ant-design/ant-design/blob/82b49c83139f481f127fe649e28ff2138a34063e/components/select/index.tsx#L147-L159

**Reply by corinne-nagel** [reply]
No worries, I'll add a ticket for this. Thanks for sharing!
